pkg/tsdb/cloud-monitoring/cloudmonitoring.go

```diff
@@ -595,27 +577,27 @@ func (s *Service) getDefaultProject(ctx context.Context, dsInfo datasourceInfo)
 	return dsInfo.defaultProject, nil
 }
 
-func unmarshalResponse(logger log.Logger, res *http.Response) (cloudMonitoringResponse, error) {
+func unmarshalResponse(res *http.Response) (cloudMonitoringResponse, error) {
 	body, err := io.ReadAll(res.Body)
 	if err != nil {
 		return cloudMonitoringResponse{}, err
 	}
 
 	defer func() {
 		if err := res.Body.Close(); err != nil {
-			logger.Warn("Failed to close response body", "err", err)
+			backend.Logger.Warn("Failed to close response body", "err", err)
```

Comment:
The logger before was populated with the `context` information. `backend.Logger` is just the default logger. So I think you need to:

- Instantiate the logger as done with testdata: https://github.com/grafana/grafana/blob/main/pkg/tsdb/grafana-testdata-datasource/testdata.go#L34 
- Then, in each method, use `s.logger.FromContext` to get a custom logger with contextual info (as it was done before).

After that, if you compare log entries, they should be similar before and after the change.

Reply:
This is still not fully solved, you need to replace all `backend.Logger` appearances and `QueryData` was using `.FromContext` before so we should probably keep it.

Reply:
Thanks @andresmgot. I replaced `backend.Logger` and added back in the context as well as updated the corresponding tests.